commands/operator-sdk/cmd/build.go

```diff
@@ -167,6 +170,27 @@ func buildFunc(cmd *cobra.Command, args []string) error {
 
 	log.Infof("Building Docker image %s", baseImageName)
 
+	dbArgs := []string{"build", ".", "-f", "build/Dockerfile", "-t", baseImageName}
+
+	if dockerBuildArgs != "" {
+		splitArgs := strings.Fields(dockerBuildArgs)
+		dbcmd := exec.Command("docker", append(dbArgs, splitArgs...)...)
+		if err := projutil.ExecCmd(dbcmd); err != nil {
+			if enableTests {
+				return fmt.Errorf("failed to output intermediate image %s: (%v)", image, err)
+			}
+			return fmt.Errorf("failed to output build image %s: (%v)", image, err)
+		}
+	} else {
+		dbcmd := exec.Command("docker", dbArgs...)
+		if err := projutil.ExecCmd(dbcmd); err != nil {
+			if enableTests {
+				return fmt.Errorf("failed to output intermediate image %s: (%v)", image, err)
+			}
+			return fmt.Errorf("failed to output build image %s: (%v)", image, err)
+		}
+	}
+
 	dbcmd := exec.Command("docker", "build", ".", "-f", "build/Dockerfile", "-t", baseImageName)
```

Comment:
Are we running the build twice here?

Reply:
we are, has been refactored according to the below comment.